Support secondary text in ListItem widget

aciccarello opened the pull request:
**Type:** bug / feature

The following has been addressed in the PR:

* [x] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**
This adds secondary text to the ListItem widget. Because secondar text often requires more height, I've replaced the single height css class with a range of sizes. This also allows different size leading/trailing content. It's a breaking change for anyone overriding that CSS class but should allow greater flexibility.

Resolves #1688

![secondary list item material theme](https://user-images.githubusercontent.com/11273838/112203216-87eed700-8bcf-11eb-9065-51a398902ae0.png)
![secondary list item dojo theme](https://user-images.githubusercontent.com/11273838/112203221-89200400-8bcf-11eb-9150-c5ed70a06bec.png)
![primary list item material theme](https://user-images.githubusercontent.com/11273838/112203227-89b89a80-8bcf-11eb-9694-5db6cca2deef.png)
![primary list item dojo theme](https://user-images.githubusercontent.com/11273838/112203230-8a513100-8bcf-11eb-8a52-a5bbee792904.png)

aciccarello committed:
Support secondary text in ListItem widget

closes #1688

tomdye commented on src/list/index.tsx:
```diff
@@ -153,9 +158,25 @@ export const ListItem = listItemFactory(function ListItem({
+		trailing = undefined
+	} = isRenderResult(firstChild) ? { primary: [firstChild, ...otherChildren] } : firstChild;
+	const size = height || (secondary ? 'medium' : 'small');
+	const sizeStyle =
```
Im not sure we should be setting a size at all, the list items should adapt to the size of their content and render appropriately. In any case we certainly shouldn't be using a massive ternary such as this.

tomdye reviewed (changes requested):
@aciccarello It seems to me that this would be better implemented as different List Item types. ie. Avatar Item, Two Line Item etc. Otherwise we're just going to end up with a large number of different properties when we could have had easy to use / reason with widgets.
It would also be good to have an example of each of these item types.

aciccarello committed:
use padding and a new class for two line list items

this makes the height more flexible
also rename main content wrapper class to .text since it wraps primary and secondary.
The leading content is given a min-width of 40px
For 24px leading icons, they will align to the front
This does not align leading icons to the top right on two-line list items
This is because the top/bottom padding is set to 8px instead of 16px
